akka-actor-typed/src/main/scala/akka/actor/typed/SpawnProtocol.scala

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed
+
+import akka.actor.typed.scaladsl.Behaviors
+
+object SpawnProtocol {
+
+  object Spawn {
+    /**
+     * Special factory to make using Spawn with ask easier
+     */
+    def apply[T](behavior: Behavior[T], name: String, props: Props): ActorRef[ActorRef[T]] ⇒ Spawn[T] =
+      replyTo ⇒ new Spawn(behavior, name, props, replyTo)
+  }
+
+  final case class Spawn[T](behavior: Behavior[T], name: String, props: Props, replyTo: ActorRef[ActorRef[T]])
+    extends SpawnProtocol {
+
+  }
+
+  val behavior: Behavior[SpawnProtocol] =
+    Behaviors.receive { (ctx, msg) ⇒
+      msg match {
+        case Spawn(bhvr, name, props, replyTo) ⇒
+          val ref =
+            if (name == null || name.equals(""))
+              ctx.spawnAnonymous(bhvr, props)
+            else
+              ctx.spawn(bhvr, name, props)
+          replyTo ! ref
+          Behaviors.same
+      }
+    }
+
```

Comment:
We also wanted to have a utility like
```
def spawn[T](parent: ActorRef[SpawnProtocol[T], b: Behavior[T], name: String, props: Props): ActorRef[T]
```

That would use something like the old [FutureActorRef](https://github.com/akka/akka/blob/f7b3b483a8cdc2fd89b07bede6997db419d30c47/akka-actor-typed/src/main/scala/akka/actor/typed/internal/ActorRefImpl.scala#L172) but I wonder if that is possible? How would we serialize such a thing? `RepointableActorRef` gets away with it because it is the actor ref that is registered in the tree and can be serialized as path string (incl uid)
.

Reply:
Did we issueify this?

Reply:
yes, that's #25305